wvlet-ui-main/src/main/scala/wvlet/lang/ui/editor/WvletMonarchLanguage.scala

```diff
@@ -138,61 +139,124 @@ val WvletMonarchLanguage: IMonarchLanguage =
+              js.Array(
+                js.RegExp("[A-Z][a-zA-Z_0-9][a-zA-Z_\\.0-9]*"),
+                new:
+                  val token = "type.identifier"
+              ),
+              js.Array(
+                js.RegExp("--.*"),
+                new:
+                  val token = "comment"
+              ),
+              js.Array(
+                js.RegExp("\"\"\""),
+                new:
+                  val token = "comment"
+                  val next  = "@multilinecomment"
+              ),
+              js.Array(
+                js.RegExp("@symbols"),
+                new:
+                  val cases = js.Dictionary("@operators" -> "operator", "@default" -> "")
+              ),
+              js.Array(
+                js.RegExp("\""),
+                new:
+                  val token = "string.quote"
+                  val next  = "@string"
+              ),
+              js.Array(
+                js.RegExp("'.*?'"),
+                new:
+                  val token = "string"
+              ),
+              js.Array(
+                js.RegExp("`.*?`"),
+                new:
+                  val token = "string.backquoted"
               )
-          ),
+            ) ++ numberTokenRules
+        val multilinecomment: Rules = js.Array(
           js.Array(
-            js.RegExp("[A-Z][a-zA-Z_0-9][a-zA-Z_\\.0-9]*"),
+            js.RegExp("\"\"\""),
             new:
-              val token = "type.identifier"
+              val token = "comment"
+              val next  = "@pop"
           ),
           js.Array(
-            js.RegExp("--.*"),
+            js.RegExp("."),
             new:
               val token = "comment"
-          ),
+          )
+        )
+        val string: Rules = js.Array(
           js.Array(
-            js.RegExp("@symbols"),
+            js.RegExp("\""),
             new:
-              val cases = js.Dictionary("@operators" -> "operator", "@default" -> "")
+              val token = "string.quote"
+              val next  = "@pop"
           ),
           js.Array(
-            js.RegExp("\".*?\""),
+            js.RegExp("\\$\\{"),
             new:
-              val token = "string"
+              val token = "string.interpolation"
+              val next  = "@interpolation"
           ),
           js.Array(
-            js.RegExp("'.*?'"),
+            js.RegExp("."),
             new:
               val token = "string"
-          ),
-          js.Array(
-            js.RegExp("`.*?`"),
-            new:
-              val token = "string.backquoted"
-          ),
-          js.Array(
-            js.RegExp("[1-9][0-9_]*.[0-9]+"),
-            new:
-              val token = "number.float"
-          ),
-          js.Array(
-            js.RegExp("0[xX][0-9a-fA-F_]+"),
-            new:
-              val token = "number"
-          ),
-          js.Array(
-            js.RegExp("[1-9][0-9_]*"),
-            new:
-              val token = "number"
           )
```

Comment:
Sorry for this hard to read diff 🙏 
These rules are declared as `numberTokenRules`.